client/src/modules/list/components/Items/ListItemDescription/ListItemDescription.jsx

```diff
@@ -0,0 +1,275 @@
+    const { description: previousDescription } = this.props;
+    const { descriptionInputValue } = this.state;
+
+    const isDescriptionUpdated = !_isEqual(
+      _trim(previousDescription),
+      _trim(descriptionInputValue)
+    );
+
+    this.setState({ isDescriptionUpdated });
+  };
+
+  renderSaveButton = () => {
+    const { disabled } = this.props;
+    const { pending } = this.state;
+
+    return (
+      <div className="list-item-description__save-button">
+        <button
+          className="primary-button"
+          disabled={disabled || pending}
+          onClick={this.handleDescriptionUpdate}
+          type="button"
+        >
+          Save
+          {pending && (
+            <Preloader
+              size={PreloaderSize.SMALL}
+              theme={PreloaderTheme.LIGHT}
+            />
+          )}
+        </button>
+      </div>
+    );
+  };
+
+  renderDescription = () => {
+    const { description, disabled } = this.props;
+    const { pending } = this.state;
+
+    return (
+      <Linkify
+        properties={{
+          target: '_blank',
+          className: 'list-item-description__link'
+        }}
+      >
+        <p
+          className={classNames('list-item-description__description', {
+            'list-item-description__description--disabled': disabled || pending
+          })}
+        >
+          {description}
+        </p>
+      </Linkify>
+    );
+  };
+
+  renderAddButton = () => {
+    const { disabled } = this.props;
+
+    return (
+      <button
+        className="list-item-description__add-button"
+        disabled={disabled}
+        onClick={this.handleShowTextarea}
+        type="button"
+      >
+        Add Description
+      </button>
+    );
+  };
+
+  renderTextarea = () => {
+    const { descriptionInputValue, pending } = this.state;
+    const { disabled } = this.props;
+
+    return (
+      <Textarea
+        className="list-item-description__edit-field"
+        disabled={disabled || pending}
+        inputRef={this.descriptionTextarea}
+        maxRows={5}
+        minRows={1}
+        name="description"
+        onBlur={this.handleBlur}
+        onChange={disabled ? null : this.handleDescriptionChange}
+        onFocus={this.handleFocus}
+        type="text"
+        value={descriptionInputValue}
+      />
+    );
+  };
+
+  handleDescriptionChange = event => {
+    const {
+      target: { value }
+    } = event;
+
+    this.setState({ descriptionInputValue: value });
+  };
+
+  render() {
+    const { isDescriptionUpdated, isTextareaVisible } = this.state;
+    const { description, disabled } = this.props;
+
+    const isSaveButtonVisible = isDescriptionUpdated && !disabled;
+    const isTitleVisible = description || isTextareaVisible;
+
+    return (
+      <Fragment>
+        <div onClick={disabled ? null : this.handleShowTextarea}>
+          {isTitleVisible && (
+            <h2 className="list-item-description__title">Description</h2>
```

Comment:
I think the title should be always visible.
After adding the description it's not clear what it is

Reply:
Hmm... The title is displayed always when there is a description or the textarea is visible. It is hidden only when there is no description and add-button is visible. So I think it's clear enough...